### ISSUE TITLE
Error with model.fit when input is a dataset

### ISSUE THREAD

**Issue by dmus** [tensorflow/tensorflow#22727]
EDIT: Probably solved, this is because no label information is provided. However, the error message is not very clear

### System information
- Tensorflow 1.10 and tf-nightly-gpu-1.12.0.dev20181004
- Ubuntu 16.04

model.fit() gives an error when the input data is a tensor that is concatenated from different datasets.

With this code example the error can be reproduced:
```
import tensorflow as tf
import numpy as np

if __name__ == '__main__':
    
    model = tf.keras.Sequential()
    model.add(tf.keras.layers.Dense(32, input_shape=(48,48,3)))
    model.add(tf.keras.layers.Dense(32))

    model.summary()

    # Compile with optimizer, loss and metrics
    optimizer = tf.train.AdamOptimizer(learning_rate=0.01)
    model.compile(loss=tf.keras.losses.mean_squared_error,
                  optimizer=optimizer)

    # make a dataset from a numpy array
    x = np.random.sample((1000,48,48,3))
    match = tf.data.Dataset.from_tensor_slices(x)

    classes = []
    for c in range(5):
        # In a real application here a dataset.filter() is used
        class_data = match
        class_data = class_data.batch(4)
        classes.append(class_data)
    
    data = list(map(lambda x: x.make_one_shot_iterator().get_next(), classes))
    batch = tf.concat(axis=0, values=data)

    model.fit(
        batch,
        steps_per_epoch=100,
        epochs=100)
```
The error is

> myproject/nightly-env/lib/python3.6/site-packages/tensorflow/python/keras/engine/training_arrays.py", line 137, in fit_loop
>     if issparse is not None and issparse(ins[i]) and not K.is_sparse(feed[i]):
> IndexError: list index out of range

The code can be even simplified further:
```
import tensorflow as tf
import numpy as np

if __name__ == '__main__':
    
    model = tf.keras.Sequential()
    model.add(tf.keras.layers.Dense(32, input_shape=(48,48,3)))
    model.add(tf.keras.layers.Dense(32))

    model.summary()

    # Compile with optimizer, loss and metrics
    optimizer = tf.train.AdamOptimizer(learning_rate=0.01)
    model.compile(loss=tf.keras.losses.mean_squared_error,
                  optimizer=optimizer)

    # make a dataset from a numpy array
    x = np.random.sample((1000,48,48,3))
    match = tf.data.Dataset.from_tensor_slices(x)
    match = match.batch(4)

    model.fit(
        match,
        steps_per_epoch=100,
        epochs=100)
```
This gives the same error

**Comment by dmus** [comment]
Issue is related to https://github.com/tensorflow/tensorflow/issues/21894.
However, this shows it is not fixed

**Comment by omalleyt12** [comment]
@dmus yep it looks like this is bc you need to provide label data, but you're right the error message should be clearer

**Comment by tensorflowbutler** [comment]
Nagging Assignee @omalleyt12: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @omalleyt12: It has been 46 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @omalleyt12: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by ymodak** [comment]
This is fixed with latest tf-nightly version '1.15.0-dev20190808'. The error message has been updated. For instance;
```python
ValueError: Arguments and signature arguments do not match. got: 17, expected: 18 `
```
